### ISSUE TITLE
Failing on photos without ICC profile embedded

### ISSUE THREAD

**Issue by kotnik** [wavexx/fgallery#26]
Processing photos stops after this error:

```
reading completed
[tificc fatal error]: Input profile is not operating in proper color space
error: command "tificc -t0 test_galerija/files/01178_chicagoatnight_1280x800.jpg.tmp test_galerija/files/01178_chicagoatnight_1280x800.jpg.tmp.tmp" failed
```

**Comment by wavexx** [comment]
Could you provide (even privately) the source image?

**Comment by wavexx** [comment]
Ping?

**Comment by kotnik** [comment]
Here it is, that exact image: https://bin.kotur.org/FcUZoU.jpg

PS. Sorry for late reply.

**Comment by Baughn** [comment]
Got the same problem, with a number of pictures.

The commonality I noticed is that they're all greyscale. Here's an example: http://krytykal.org/wp-content/uploads/2014/11/OSO_v03_275.jpg

**Comment by wavexx** [comment]
On 31/12/15 18:33, Svein Ove Aas wrote:
> Got the same problem, with a number of pictures.
> 
> The commonality I noticed is that they're all greyscale. Here's an
> example: http://krytykal.org/wp-content/uploads/2014/11/OSO_v03_275.jpg

Do you know how the greyscale conversion was performed?
In-camera maybe?

I debugged this the last time. It looks like that after the image has
been converted to greyscale, the ICC profile was left to whatever was
before (ie: not a single-channel profile).

I'm unsure whether the correct approach would be to remap in RGB and
back, or drop the ICC profile entirely. Probably the second.

**Comment by Baughn** [comment]
@wavexx If you look at the picture, you'll realise it wasn't made with a camera.

I don't know how the picture was created, though; I'm just the one who wants them in a gallery. IMO fgallery should work with any picture the browser is willing to display.

**Comment by wavexx** [comment]
On 01/01/16 16:29, Svein Ove Aas wrote:
> @wavexx <https://github.com/wavexx> If you look at the picture, you'll
> realise it wasn't made with a camera.
>
> I don't know how the picture was created, though; I'm just the one who
> wants them in a gallery. IMO fgallery should work with any picture the
> browser is willing to display.

Sure, I was trying to always preserve the profile information where
possible. I would ignore tificc only when really necessary (possible never).

I now check also for the ColorSpace exif tag to prevent calling tificc
for pre-converted images, which fixes both images. In fact, for some of
my test galleries, only a few images actually require tificc, with a
nice performance gain.

Could you test it?

**Comment by Baughn** [comment]
Works great. ^_^

**Comment by wavexx** [comment]
On 02/01/16 18:43, Svein Ove Aas wrote:
> Works great. ^_^

Then I give you the honor of closing the bug ;)

**Comment by Baughn** [comment]
I'm pretty sure I can't. There's no "close" button here. Um, this comment fixes #26?

I'd be amazed if that works.

**Comment by wavexx** [comment]
Hum, maybe you're not the original reporter?
Anyway, closing™
